lib/Validators/BaseValidator.php

```diff
@@ -0,0 +1,136 @@
+<?php
+declare(strict_types=1);
+
+namespace OCA\Deck\Validators;
+
+use Exception;
+use OCA\Deck\BadRequestException;
+
+abstract class BaseValidator {
+
+	private const MAX_LENGTH = 255;
+
+	/**
+	 * @return array
+	 */
+	abstract function rules();
+
+	/**
+	 * Validate given entries
+	 *
+	 * @param array $data
+	 * @return bool
+	 */
+	private function validate ($data) {
+		$rules = $this->rules();
+
+		foreach ($data as $field => $value) {
+			$field_rule = $rules[$field];
+
+			if (is_array($field_rule)) {
+				foreach ($field_rule as $rule) {
+					// The format for specifying validation rules and parameters follows an
+					// easy {rule}:{parameters} formatting convention. For instance the
+					// rule "Max:3" states that the value may only be three letters.
+					if (strpos($rule, ':') !== false) {
+						[$rule, $parameter] = explode(':', $rule, 2);
+						if (!$this->{$rule}($value, $parameter)) {
+							throw new BadRequestException(
+								$this->getErrorMessage($rule, $field, $parameter));
+						}
+
+					} else {
+						if (!$this->{$rule}($value)) {
+							throw new BadRequestException(
+								$field . ' must be provided and must be '. str_replace("_", " ", $rule));
+						}
+					}
+				}
+			}
+
+			if (is_callable($field_rule) && !$field_rule($value)) {
+				throw new BadRequestException($field . ' must be provided');
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * @param array $data
+	 * @return bool
+	 */
+	public function check(array $data)
+	{
+		return $this->validate($data);
+	}
```

Comment:
```suggestion
	public function check(array $data): bool {
		return $this->validate($data);
	}
```

Reply:
Actually it seems to throw so maybe rather use void and annotate the exception that is being thrown here.